sunpy/image/transform.py

```diff
@@ -22,6 +24,7 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
 
     This function supports NaN values in the input image and supports using NaN for
     pixels in the output image that are beyond the extent of the input image.
+    Handling NaN values in the input image requires additional computation time.
```

Comment:
I am kind of amazed that no affine transform implementation just handles nans :(

Reply:
The scikit-image interpolation orders that are handled by native Cython code – nearest-neighbor, bi-linear, and bi-cubic – do handle image NaNs.  However, the current implementation is an all-or-nothing approach; since half of the allowed orders do not handle NaNs (the orders that call SciPy code), we can't mark scikit-image as `handles_image_nans=True`.

SciPy's `affine_transform()` is actually two steps: a spline pre-filtering step and a polynomial interpolation.  The polynomial interpolation handles image NaNs fine.  The problem is the spline pre-filtering, which turns the entire image to NaNs if any pixel is NaN.  And, apparently, that is formally correct: when it comes to spline pre-filtering, every pixel in the image contributes at some level to the resulting spline interpolation, no matter how far away that pixel is.